### DISCUSSION TITLE
Unable to INSERT

### DISCUSSION THREAD

**Discussion by murraycollingwood** [propelorm/Propel2#1870] [Q&A] [answered]
Another "new to Propel2" issue:
```
Uncaught exception "Propel\Runtime\Exception\PropelException" with message "Unable to execute INSERT statement [INSERT INTO sarlog (id, schoolid, userid, timelock, studid, markabsent, code, explainid, explain, otherinfo, absentid, classid) VALUES (:p0, :p1, :p2, :p3, :p4, :p5, :p6, :p7, :p8, :p9, :p10, :p11)]" in /home/sobs/generated-classes/Base/Sarlog.php on line 1187 school=1 username=2
Context: 0 = Array ( [0] => Array ( [file] => /home/sobs/generated-classes/Base/Sarlog.php [line] => 1059 [function] => doInsert [class] => Base\Sarlog [type] => -> [args] => Array ( [0] => Propel\Runtime\Connection\ConnectionWrapper Object ( [name:protected] => focus_sobs [useDebug] => [connection:protected] => Propel\Runtime\Connection\PdoConnection Object ( [name:protected] => [pdo:protected] => PDO Object ( ) ) [nestedTransactionCount:protected] => 0 [isUncommitable:protected] => ...
```
This is a table of log entries - we only ever insert records into this table.  The insert interactions all look like this:
```
		$log = new Sarlog();
		$log->setSchoolid($currentschool->getId());
		$log->setUserid($currentuser->getId());
		$log->setTimelock(time()); // Server time
		$log->setOtherinfo($source);
		$log->setClassid($sarclass->getId());
		$log->save();
```

I've been through the other sections of code and can't find any locations where I have referenced of set the ID field (I've been using Propel1.6 for many years, and this code has been working fine up until now).

Any ideas why this might be happening?

Cheers
Murray

**Comment by murraycollingwood** [comment, marked as answer]
Update: I'm recalling a similar error, something to do with fields that clash with the MySQL reserved words list.
I have two, "code" and "explain".
I will try changing these.

**Reply by murraycollingwood** [reply]
Yep - that was it!

**Comment by mringler** [comment]
Nice, you found it.
But it sounds like there was some guesswork going on, which should not be necessary. You can get better error messages by activating logging and debug mode, see http://propelorm.org/documentation/07-logging.html. I think those should have given you the actual error message from the database.
